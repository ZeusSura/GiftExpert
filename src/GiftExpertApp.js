import React, { useState } from "react";
import CategoryAdd from "./components/CategoryAdd";
import GridGift from "./components/GirdGift";
import shortid from "shortid";
const GiftExpertApp = ({defaultCategories =[]}) => {
  const [categorias, setCategorias] = useState(defaultCategories);

  return (
    <>
      <h2>GiftExpertApp</h2>
      <CategoryAdd setCategorias={setCategorias} />
      <hr />
      {categorias.map((categoria) => (
        <GridGift key={shortid.generate()} categoria={categoria} />
      ))}
    </>
  );
};

export default GiftExpertApp;
