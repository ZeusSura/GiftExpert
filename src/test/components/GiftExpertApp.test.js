import React from 'react'
import GiftExpertApp from '../../GiftExpertApp'
import {shallow } from 'enzyme'

describe('Pruebas sobre el componente GiftExpertApp',()=>{




    test('Deberia hacer mach con el snapshots', () => {
        let  wrapper =shallow(<GiftExpertApp/>)
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe mostrar una lista de categorias', () => {
        const categories = ['dragon ball','one punch']

        const  wrapper =shallow(<GiftExpertApp defaultCategories={categories}/>)

        expect(wrapper.find('GridGift').length).toBe(categories.length)
    })
    
})
