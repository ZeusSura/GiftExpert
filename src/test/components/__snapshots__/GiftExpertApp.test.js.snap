// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GiftExpertApp Deberia hacer mach con el snapshots 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <CategoryAdd
    setCategorias={[Function]}
  />
  <hr />
</Fragment>
`;
